### DISCUSSION TITLE
Grid days sensor faulted, so-status says 5x5

### DISCUSSION THREAD

**Discussion by sempervictus** [Security-Onion-Solutions/securityonion#8417] [Unsupported Versions]
Grid says that the process status on a sensor is unknown, sensor shows all green on `so-status`; no idea how the stack interprets the latter into the former (or if its even using `so-status` in the new hotness) but it seems incongruent:
![image](https://user-images.githubusercontent.com/1331084/182024471-3b68cc54-ad29-47c8-a228-83dde40b24f9.png)
![image](https://user-images.githubusercontent.com/1331084/182024500-78ab29dd-e225-4620-9e9c-41a338ba1a9c.png)
The head-end showed this same thing for about a day, didnt show the sensor at all for the first few hours after joining, now shows the sensor as faulted without much reasoning.

**Comment by dougburks** [comment]
Sometimes it can take some time to change from `Fault` to `OK`. Is it still showing `Fault` today?

Have you checked logs in `/opt/so/log/` for additional clues?

Is the sensor able to connect to the manager on all ports listed at https://docs.securityonion.net/en/2.3/firewall.html#node-communication?

**Comment by sempervictus** [comment]
@dougburks thanks for picking this up. Still showing fault status today, but EPS counters are fine, data is flowing fine, and all indicators on the sensor itself say that its happy. Is there any mechanism by which to reset the state in the grid to "all is well" and then observe what trips a state change to showing faults if such occurs?

**Comment by dougburks** [comment]
Is the sensor able to connect to the manager on all ports listed at https://docs.securityonion.net/en/2.3/firewall.html#node-communication?

If you run the following command on the manager, do you see a firewall rule that allows the sensor to connect to port 443?
```
sudo iptables -nvL |grep 443
```

If you run tcpdump on the management interface of the manager, do you see traffic from the sensor connecting to port 443?

On the sensor, what is the output of the following?
```
sudo so-status -q; echo $?
```

**Comment by sempervictus** [comment]
On the server i do have 443 open to the sensor:
```
# sudo iptables -nvL |grep 443
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.5           tcp dpt:443
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.4           tcp dpt:4434
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.4           tcp dpt:4433
 3411  207K ACCEPT     tcp  --  *      *       10.0.0.0/8           0.0.0.0/0            tcp dpt:443
```
On the sensor i get 99:
```
# so-status -q; echo $?
99
```
i _am_ very much seeing data from Zeek and Suricata though, so the important pieces are working

**Reply by dougburks** [reply]
so-status runs the following:
```
SYSTEM_START_TIME=$(date -d "$(</proc/uptime awk '{print $1}') seconds ago" +%s)
LAST_SOSETUP_LOG=$([ -e "/root/sosetup.log" ] && date -r /root/sosetup.log +%s || echo 0)
<snip>
if [ $SYSTEM_START_TIME -lt $LAST_SOSETUP_LOG ]; then
   exit 99
fi
```

Did you reboot the sensor after setup completed?

If so, what is the output of the following on the sensor?
```
SYSTEM_START_TIME=$(date -d "$(</proc/uptime awk '{print $1}') seconds ago" +%s)
LAST_SOSETUP_LOG=$([ -e "/root/sosetup.log" ] && date -r /root/sosetup.log +%s || echo 0)
echo $SYSTEM_START_TIME
echo $LAST_SOSETUP_LOG
```

**Comment by sempervictus** [comment]
"Reboot" is kind of a convoluted term in Linux these days - the LXC container in which the SO sensor runs has been rebooted, but not the  physical host (has other duties).
The two variables generated output to:
```
1659135519
1659188235
```
I'm not sure how `/proc/uptime` works in this context - not sure whether LXC maps a faux value for the container's startup there or if it just uses the host's uptime.

**Comment by dougburks** [comment]
I suppose you could always change the timestamp on `/root/sosetup.log` and make it earlier than 1659135519.

**Comment by sempervictus** [comment]
Thanks, will try that this evening and mark resolved if that works.

**Comment by sempervictus** [comment]
Clever hack, but unfortunately didn't quite do the trick.
On the bright side, i've got a scheduled physical downtime tomorrow for getting a pair of 100G TAP ports into the thing so physical host uptime should shift appropriately.

**Reply by TOoSmOotH** [reply]
Is there data in grafana for this host? If not its an influx connectivity issue.
